python/shotgun_model/data_handler_find.py

```diff
@@ -0,0 +1,384 @@
+                                          ``[{"preset_name":"LATEST","latest_by":"BY_PIPELINE_STEP_NUMBER_AND_ENTITIES_CREATED_AT"}]``
+        :param cache_path:                Path to cache file location
+        :param parent:                    Parent QT object
+        """
+        super(ShotgunFindDataHandler, self).__init__(cache_path, parent)
+        self.__entity_ids = None
+        self.__entity_type = entity_type
+        self.__filters = filters
+        self.__order = order
+        self.__hierarchy = hierarchy
+        self.__fields = fields
+        self.__download_thumbs = download_thumbs
+        self.__limit = limit
+        self.__additional_filter_presets = additional_filter_presets
+
+    def _clear_cache(self):
+        """
+        Sets up an empty cache in memory
+        """
+        self.__entity_ids = None
+        super(ShotgunFindDataHandler, self)._clear_cache()
+
+    def unload_cache(self):
+        """
+        Unloads any in-memory cache data.
+        """
+        self.__entity_ids = None
+        super(ShotgunFindDataHandler, self).unload_cache()
+
+    def get_entity_ids(self):
+        """
+        Returns a list of entity ids contained in this data set given an entity type.
+
+        :return: A list of unique ids for all items in the model.
+        :rtype: ``list``
+        """
+        # loop over all cache items - the find data handler organizes
+        # its unique ids so that all leaf nodes (e.g. representing an entity
+        # are ints and all other items are strings
+
+        # memoized for performance
+        if self.__entity_ids is None:
+            entity_ids = []
+            for uid in self._cache[self.CACHE_BY_UID].keys():
+                if isinstance(uid, int):
+                    # this is a leaf node representing an entity
+                    entity_ids.append(uid)
+            self.__entity_ids = entity_ids
+
+        return self.__entity_ids
+
+    def get_uid_from_entity_id(self, entity_id):
+        """
+        Returns the unique id for a given entity or None if not found
+
+        :param entity_id: Shotgun entity id to resolve
+        :returns: unique id as string or int, to be used with
+                  :meth:`get_data_item_from_uid`.
+        """
+        for uid in self._cache[self.CACHE_BY_UID].keys():
+            if isinstance(uid, int) and uid == entity_id:
+                return uid
+        return None
+
+    def generate_data_request(self, data_retriever):
+        """
+        Generate a data request for a data retriever.
+
+        Once the data has arrived, the caller is expected to
+        call meth:`update_data` and pass in the received
+        data payload for processing.
+
+        :param data_retriever: :class:`~tk-framework-shotgunutils:shotgun_data.ShotgunDataRetriever` instance.
+        :returns: Request id or None if no work is needed
+        """
+        # only request data from shotgun is filters are defined.
```

Comment:
you are confusing `filters=[]` with `filters=None` - setting filters to `None` is an indication that we don't want any data to be fetched but the model to be populated with external data only!